platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/software/schemaFromSoftwareTypes.kt

```diff
@@ -87,9 +123,32 @@ data class SoftwareTypeInfo(
 
 
 private
-fun extensionConfiguringFunctions(typeToExtend: KClass<*>, softwareTypeImplementations: Iterable<SoftwareTypeInfo>): FunctionExtractor = object : FunctionExtractor {
+fun softwareTypeConfiguringFunctions(typeToExtend: KClass<*>, softwareTypeImplementations: Iterable<SoftwareTypeInfo<*>>): FunctionExtractor = object : FunctionExtractor {
+    override fun memberFunctions(kClass: KClass<*>, preIndex: DataSchemaBuilder.PreIndex): Iterable<SchemaMemberFunction> =
+        if (kClass == typeToExtend) softwareTypeImplementations.map(SoftwareTypeInfo<*>::schemaFunction) else emptyList()
+
+    override fun constructors(kClass: KClass<*>, preIndex: DataSchemaBuilder.PreIndex): Iterable<DataConstructor> = emptyList()
+
+    override fun topLevelFunction(function: KFunction<*>, preIndex: DataSchemaBuilder.PreIndex) = null
+}
+
+
+private
+fun conventionsFunction(): FunctionExtractor = object : FunctionExtractor {
     override fun memberFunctions(kClass: KClass<*>, preIndex: DataSchemaBuilder.PreIndex): Iterable<SchemaMemberFunction> =
-        if (kClass == typeToExtend) softwareTypeImplementations.map(SoftwareTypeInfo::schemaFunction) else emptyList()
+        if (kClass == ConventionsTopLevelReceiver::class) listOf(
```

Comment:
You can convert `ConventionsTopLevelReceiver` from a class to an interface and declare this function right there, it is fine to have such fake functions if we can provide the runtime implementation for them, or if we don't need to invoke them at runtime at all.

Reply:
I had originally done this, but had some difficulty with it.  I'm sure I was missing some crucial detail.  I've changed this as you suggested and removed the synthetic function member.  Thanks for pointing this out.